spmd/tensor/device_mesh.py

```diff
@@ -262,28 +276,44 @@ def broadcast(self, tensor: torch.Tensor, mesh_dim: int = 0):
         return broadcast(tensor.contiguous(), src=src_for_dim, group=dim_group)
 
     # pyre-fixme[3]: Return type must be annotated.
-    def all_gather(self, tensor: torch.Tensor, mesh_dim: int = 0):
+    def all_gather(
+        self, tensor: torch.Tensor, mesh_dim: int = 0, tensor_dim: int = 0
```

Comment:
do we really need tensor_dim in `all_gather`? I can understand we need it in `all_gather_base` to make sure reshuffling the dim or concating it in non-0 dimension. But seems we don't need it here, or is it because we want to keep the all_gather API consistent? IMO we can probably have a follow up to delete `all_gather_base` API and use this `all_gather` API with the semantic that it will directly return a big tensor instead of a tensor list :)

Reply:
Oh makes sense -- it doesn't do anything anyways!